tests/MCPClient/test_policy_check.py

```diff
@@ -0,0 +1,592 @@
+    execute_or_run: mock.Mock,
+    sip_file: models.File,
+    sip: models.SIP,
+    sip_file_format_version: models.FileFormatVersion,
+    format: fprmodels.Format,
+    format_version: fprmodels.FormatVersion,
+    shared_directory_path: str,
+) -> None:
+    job = mock.Mock(
+        args=[
+            "policy_check",
+            sip_file.currentlocation.decode(),
+            str(sip_file.uuid),
+            str(sip.uuid),
+            str(shared_directory_path),
+            sip_file.filegrpuse,
+        ],
+        JobContext=mock.MagicMock(),
+        spec=Job,
+    )
+
+    policy_check.call([job])
+
+    job.set_status.assert_called_once_with(0)
+
+    job.pyprint.assert_called_once_with(
+        "Not performing a policy check because there are no relevant FPR rules"
+    )
+
+
+@pytest.mark.django_db
+@mock.patch(
+    "policy_check.executeOrRun",
+    return_value=(
+        1,
+        STDOUT,
+        "",
+    ),
+)
+def test_policy_checker_warns_if_file_does_not_exist(
+    execute_or_run: mock.Mock,
+    sip_file: models.File,
+    sip: models.SIP,
+    sip_file_format_version: models.FileFormatVersion,
+    format: fprmodels.Format,
+    format_version: fprmodels.FormatVersion,
+    shared_directory_path: str,
+) -> None:
+    job = mock.Mock(
+        args=[
+            "policy_check",
+            "",
+            "",
+            str(sip.uuid),
+            str(shared_directory_path),
+            sip_file.filegrpuse,
+        ],
+        JobContext=mock.MagicMock(),
+        spec=Job,
+    )
+
+    policy_check.call([job])
+
+    job.set_status.assert_called_once_with(0)
+
+    job.pyprint.assert_called_once_with(
+        "Not performing a policy check because there is no file with UUID ."
+    )
+
+
+@pytest.mark.django_db
+@mock.patch(
+    "policy_check.executeOrRun",
+    return_value=(
+        1,
+        STDOUT,
+        "",
+    ),
+)
+def test_policy_checker_fails_if_execute_rule_command_fails(
+    execute_or_run: mock.Mock,
+    sip_file: models.File,
+    sip: models.SIP,
+    fprule_policy_check: fprmodels.FPRule,
+    sip_file_format_version: models.FileFormatVersion,
+    format: fprmodels.Format,
+    format_version: fprmodels.FormatVersion,
+    shared_directory_path: str,
+) -> None:
+    job = mock.Mock(
+        args=[
+            "policy_check",
+            sip_file.currentlocation.decode(),
+            str(sip_file.uuid),
+            str(sip.uuid),
+            str(shared_directory_path),
+            sip_file.filegrpuse,
+        ],
+        JobContext=mock.MagicMock(),
+        spec=Job,
+    )
+
+    policy_check.call([job])
+
+    job.set_status.assert_called_once_with(1)
+    assert job.pyprint.mock_calls == [
+        mock.call("Running", fprule_policy_check.command.description)
+    ]
+    job.print_error.assert_called_once_with(
+        f"Command {fprule_policy_check.command.description} failed with exit status 1; stderr:",
+        "",
```

Comment:
This could reuse the `stderr` value from the mock if you set its `return_value` inside the test.